File: backend/src/models/invoices/book_invoice.rb
require_relative './invoice'

class BookInvoice < Invoice
  def initialize order:
    super
    generate_shipping_label
    send_notification
  end

  def generate_shipping_label
    puts "Gerei o 'shipping label' para o endereco #{order.shipping_address.zipcode}"
  end

  def send_notification
    puts "Notifquei de insencao de impostos no email #{order.customer.email}"
  end
end

File: src/main/kotlin/challange/order/invoices/BookInvoice.kt
class BookInvoice(order: Order, orderItem: List<OrderItem>) : Invoice(order, orderItem) {

    override fun generateShippingLabel(): String {
        return "Envio de livro para ${shippingAddress.toString()} - Item isento de impostos conforme disposto na Constituição Art. 150, VI, d."
    }

}